feature-utils/poly-look/src/react-components/legend.js

```diff
@@ -0,0 +1,50 @@
+class IncorrectLegendEntry extends Error {
+  constructor(entry) {
+    super(`«${JSON.stringify(entry)}» is not a LegendEntry`);
+  }
+}
+
+class LegendEntry {
+  constructor(description, color) {
+    this._description = description;
+    if (!color.match(/^#[0-9a-fA-F]{6}$/)) {
+      throw new IncorrectLegendEntry({
+        description: description,
+        color: color,
+      });
+    }
+    this._color = color;
+  }
+
+  get description() {
+    return this._description;
+  }
+
+  get color() {
+    return this._color;
+  }
+}
+
+class Legend {
+  constructor(legends) {
+    let thisLegends = [];
+    for (const legend of legends) {
+      if (legend.constructor.name === "LegendEntry") {
+        thisLegends.push(legend);
+        return;
+      }
+      if ("description" in legend && "color" in legend) {
```

Comment:
doesn't this imply that in this case `legend` is instanceOf `LegendEntry`, aka enters the 1st if?Can it happen that is sth else? 🤔

Reply:
Right, `legend` (formerly `l`) is an object that can be transformed in `LegendEntry`; not a `LegendEntry` yet. Maybe the naming is a bit misleading.